_posts/en/newsletters/2024-01-03-newsletter.md

```diff
@@ -0,0 +1,402 @@
+  Ingala suggests a way those users could effect a single exit with a
+  single transaction that could have a single input and five outputs,
+  giving them the typical [payment batching][topic payment batching]
+  reduction in transaction size by about 50%.
+
+  In complex multiparty contracts with very large numbers of users, the
+  reduction in onchain size can easily be significantly larger than 50%.
+  Even better, if the five active users simply wanted to move their
+  funds to a new shared UTXO involving just them, they could use a
+  single-input and single-output transaction, saving about 80% in the
+  case of five users or about 99% in the case of a hundred users.  That
+  huge savings for large groups of users moving their funds from one
+  contract to another may be critical when transaction feerates are high
+  and many users have relatively small balances in the contract. For
+  example 100 users each have a balance of 10,000 sats ($4 USD at the
+  time of writing); if they each individually had to pay transaction
+  fees to exit the contract and enter a new contract, then even with an
+  improbably small spending transaction size of 100 vbytes, a
+  transaction fee of 100 sats/vbyte would consume their entire balance.
+  If they are able to move their combined funds of 1 million sats in a
+  single 200 vbyte transaction at 100 sats/vbyte, then each individual
+  user will only pay 200 sats (2% of their balance).
+
+    The payment batching is accomplished by having one of the
+    participants in the multiparty contract protocol construct a spend
+    of the shared funds to the outputs agreed upon by the other active
+    participants.  The contract allows this---but only if the
+    constructing party funds a bond that they will forfeit if anyone can
+    prove that they spent the contract protocol's funds incorrectly; the
+    bond amount should be considerably more than the constructing party
+    can gain from attempting an incorrect transfer of funds.  If no one
+    generates a fraud proof showing the constructing party acted
+    inappropriately within a period of time, the bond is refunded to the
+    constructing party.  Ingala roughly describes how this feature could
+    be added to a multiparty contract protocol using [OP_CAT][],
+    `OP_CHECKCONTRACTVERIFY`, and amount introspection from the proposed
+    [MATT][] soft fork, with him noting that it would be easier with the
+    addition also of [OP_CSFS][topic op_checksigfromstack] and 64-bit
+    arithmetic operators in tapscript.
+
+    The idea has received a small amount of discussion as of this
+    writing.
+
+- **New coin selection strategies:** Mark Erhardt [posted][erhardt coin]
+  to Delving Bitcoin about edge-cases users may have experienced with
+  Bitcoin Core's [coin selection][topic coin selection] strategy and
+  proposes two new strategies, plus a method of choosing the best
+  strategy from among those available.  He also summaries the benefits
```

Comment:
This sounds like it’s a new proposal, but we have been using the waste metric as a method of choosing an input set from all input set candidates for multiple major releases.

```suggestion
  proposes two new strategies.
  He also summaries the benefits
```

Perhaps it could be added that the two new algorithms both aim to reduce input set size at high feerates and thus improve the mentioned edge-cases.